projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -644,6 +646,37 @@ function ( $parent_file ) use ( $jetpack_slug ) {
 		);
 	}
 
+	/**
+	 * 1. Remove the Gutenberg plugin menu
+	 * 2. Re-add the Site Editor menu
+	 *
+	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
+	 */
+	public function add_gutenberg_menus( $wp_admin = false ) {
+		// Always remove the Gutenberg menu.
+		remove_menu_page( 'gutenberg' );
+
+		// We can bail if we don't meet the conditions of the Site Editor.
+		if ( ! ( function_exists( 'gutenberg_is_fse_theme' ) && gutenberg_is_fse_theme() ) ) {
+			return;
+		}
+
+		// Core Gutenberg registers without an explicit position, and we don't want the (beta) tag.
+		remove_menu_page( 'gutenberg-edit-site' );
+
+		$link = $wp_admin ? 'gutenberg-edit-site' : 'https://wordpress.com/site-editor/' . $this->domain;
```

Comment:
Does the `https://wordpress.com/site-editor/` URL work for Jetpack sites as well? If so, we can leave this here, but otherwise we should move these overrides to the `Atomic_Admin_Menu` and `WPcom_Admin_Menu` classes.

Reply:
I think it should work. We are using the same iframing code as for the post editor, which works for Jetpack sites.

Another minor note about this line - Hardcoding `wordpress.com` will change calypso envs. E.g. If we start with Horizon and navigate via this menu we'll be switched to Staging/Production.

Reply:
> I think it should work. We are using the same iframing code as for the post editor, which works for Jetpack sites.
👍 

> Another minor note about this line - Hardcoding wordpress.com will change calypso envs.

Yeah, that's a problem we have with the rest of menus. Actually, this is a problem that happens only when the menu is rendered on WP Admin, since we remove the `wordpress.com` domain on API requests (used by Calypso) to make the URLs relative to the Calypso env:

https://github.com/Automattic/jetpack/blob/7d0fcdfc99c9fc2a1d1ddf60858c0a292cbdf385/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php#L298-L300

Reply:
We're only focused on Simple sites at present, so I'll migrate this into `WPcom_Admin_Menu`. Although iframing will work on Jetpack and Atomic sites in theory, we can do that later when we're ready to open up testing.

Reply:
As for the `wordpress.com` issue with Calypso (or other envs), it might be nice to add a filterable URL in the class that all of the methods use.

Reply:
That sounds good. I'm also fine with leaving that for a separate PR.